node/overseer/src/lib.rs

```diff
@@ -591,23 +1003,7 @@ where
 	///
 	/// # fn main() { executor::block_on(async move {
 	/// let spawner = sp_core::testing::TaskExecutor::new();
-	/// let all_subsystems = AllSubsystems {
-	///     candidate_validation: ValidationSubsystem,
-	///     candidate_backing: DummySubsystem,
-	///     candidate_selection: DummySubsystem,
-	///     statement_distribution: DummySubsystem,
-	///     availability_distribution: DummySubsystem,
-	///     bitfield_signing: DummySubsystem,
-	///     bitfield_distribution: DummySubsystem,
-	///     provisioner: DummySubsystem,
-	///     pov_distribution: DummySubsystem,
-	///     runtime_api: DummySubsystem,
-	///     availability_store: DummySubsystem,
-	///     network_bridge: DummySubsystem,
-	///     chain_api: DummySubsystem,
-	///     collation_generation: DummySubsystem,
-	///     collator_protocol: DummySubsystem,
-	/// };
+	/// let all_subsystems = AllSubsystems::<()>::dummy().replace_candidate_validation(ValidationSubsystem);
```

Comment:
I don't understand why the `AllSubsystems::<()>::dummy()` syntax is required; I'd have imagined that `AllSubsystems::dummy()` would have sufficed. It looks like all type parameters in the struct already have defaults.

Reply:
It's a rust "bug" mentioned in https://github.com/paritytech/polkadot/pull/1794/files#diff-7493087dab6f0ff3822fcb707cf51a4fR475-R482.

Reply:
I have documented this in the docs of the `dummy` function. That is some rustc bug AFAIK